docs/content/latest/api/ysql/extensions.md

```diff
@@ -0,0 +1,361 @@
+INSERT INTO spi_test VALUES(3, 'desc3');
+
+SET ROLE yugabyte;
+INSERT INTO spi_test VALUES(4, 'desc4');
+
+SELECT * FROM spi_test ORDER BY id;
+```
+
+```
+ id | content | username |          moddate
+----+---------+----------+----------------------------
+  1 | desc1   | yugabyte | 2019-09-13 16:55:53.969907
+  2 | desc2   | postgres | 2019-09-13 16:55:53.983306
+  3 | desc3   | postgres | 2019-09-13 16:55:53.98658
+  4 | desc4   | yugabyte | 2019-09-13 16:55:53.991315
+(4 rows)
+```
+
+- Update some rows.
+
+Should update both `username`  and `moddate` accordingly.
+
+```sql
+UPDATE spi_test SET content = 'desc1_updated' WHERE id = 1;
+UPDATE spi_test SET content = 'desc3_updated' WHERE id = 3;
+
+SELECT * FROM spi_test ORDER BY id;
+```
+
+```
+ id |    content    | username |          moddate
+----+---------------+----------+----------------------------
+  1 | desc1_updated | yugabyte | 2019-09-13 16:56:27.623513
+  2 | desc2         | postgres | 2019-09-13 16:55:53.983306
+  3 | desc3_updated | yugabyte | 2019-09-13 16:56:27.634099
+  4 | desc4         | yugabyte | 2019-09-13 16:55:53.991315
+(4 rows)
+```
+
+Read more about the spi module in the [PostgreSQL Docs](https://www.postgresql.org/docs/current/contrib-spi.html).
+
+
+## Install Other Extensions
+
+Other extensions have to be installed manually into the YugaByte `lib` folder before they can be enabled (with `CREATE EXTENSION`).
+
+{{< note title="Note" >}}
+
+Currently, in a multi-node setup, the installation instructions below must be done on _every_ node in the cluster.
+
+{{< /note >}}
+
+
+Typically extensions need three types of files:
+
+- Shared library files (`<name>.so`)
+- SQL files (`<name>--<version>.sql`)
+- Control files (`<name>.control`)
+
+In order to install an extension you need to copy these files into the respective folders of your YugaByte installation.
+
+
+Shared library files will be in the `pkglibdir` directory while SQL and control files should be in the `extension` subdirectory of the 
+`libdir` directory.
+To find these direcotries on your local installation, you can use YugaByte's `pg_config` executable.
+First, alias it to `yb_pg_config`. 
+Assuming your YugaByte installation folder is `$YUGABYTE_HOME`, you can run:
+```sh
+$ alias yb_pg_config=$YUGABYTE_HOME/postgres/bin/pg_config
+```
+
+Now you can list existing shared libraries with:
+```sh
+$ ls $(yb_pg_config --pkglibdir)
+```
+
+And SQL and control files for already-installed extensions with:
+```sh
+$ ls $(yb_pg_config --sharedir)/extension/
+```
+
+To obtain these files for your target extension, you can build it from scratch following extension's build instructions.
+Alternatively, if you already have PostgreSQL (ideally version `11.2` for best YugaByte YSQL compatibility) with that extension installed, 
+then you can find these files as follows:
+
+```sh
+$ ls $(pg_config --pkglibdir) | grep <name>
+```
+
+```sh
+ls $(pg_config --sharedir)/extension/ | grep <name>
```

Comment:
You might want to wrap the `$(...)` in quotes because if it expands to something with spaces, Bash treats them as separate arguments.  Zsh doesn't seem to have this problem.  I haven't tested other shells.  Apply this to the various other places where environment variables are used.